src/bluesearch/database/article.py

```diff
@@ -73,6 +76,127 @@ def paragraphs(self) -> Iterable[Tuple[str, str]]:
         """
 
 
+class PubmedXMLParser(ArticleParser):
+    """Parser for PubMed XML files using the JATS Journal Publishing DTD.
+
+    Parameters
+    ----------
+    path
+        The path to the XML file from PubMed.
+    """
+
+    def __init__(self, path: Union[str, Path]) -> None:
+        self.content = etree.parse(str(path))  # nosec
+
+    @property
+    def title(self) -> str:
+        """Get the article title.
+
+        Returns
+        -------
+        str
+            The article title.
+        """
+        titles = self.content.find("//title-group/article-title")
+        if titles is None:
+            return ""
+        return self.text_content(titles)
+
+    @property
+    def authors(self) -> Generator[str, None, None]:
```

Comment:
So what is the main motivation behind using a `Genarator`? (Also done for `abstract` and `paragraphs`).
On the outside, it might make the user believe that there is "lazy" parsing taking place. However, I quickly checked the methods `xpath` and `findall` and they seem to return lists.


Note that this is just a minor comment and I am totally fine with the way it is

Reply:
I would say it was mainly to keep consistency with what was done in `CORD19ArticleParser`.

Reply:
Yeh, you are right that my remark also applies to the `CORD19ArticleParser`. @Stannislav your take on this?

Reply:
So in `CORD19ArticleParser` it was just for convenience to avoid caching all items in a list before returning them.

The base class `ArticleParser` has the return type `Iterable[str]`, so it's not at all necessary to use generators, anything that's iterable will do.

Reply:
I feel like what we are doing is very similar to
```python
l = [1, 2, 3, 4]  
g = (x for x in l) 
```
Seems like false advertizing for lazy evaluation and memory efficiency.